modules/text2vec-weaviate/ent/class_settings.go

```diff
@@ -0,0 +1,105 @@
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package ent
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/entities/moduletools"
+	basesettings "github.com/weaviate/weaviate/usecases/modulecomponents/settings"
+)
+
+const (
+	// TODO: replace docker internal host with actual host
+	DefaultBaseURL               = "https://embedding.labs.weaviate.io"
+	DefaultWeaviateModel         = "Snowflake/snowflake-arctic-embed-m-v1.5"
+	DefaultTruncate              = "right"
+	DefaultVectorizeClassName    = true
+	DefaultPropertyIndexed       = true
+	DefaultVectorizePropertyName = false
+)
+
+const (
+	SnowflakeArcticEmbedM = "Snowflake/snowflake-arctic-embed-m-v1.5"
+)
+
+var (
+	availableWeaviateModels = []string{
+		SnowflakeArcticEmbedM,
+	}
+	availableTruncates = []string{"left", "right"}
+)
+
+var SnowflakeArcticEmbedMDefaultDimensions int64 = 768
+
+var availableWeaviateModelsDimensions = map[string][]int64{
+	SnowflakeArcticEmbedM: {SnowflakeArcticEmbedMDefaultDimensions, 256},
+}
+
+type classSettings struct {
+	basesettings.BaseClassSettings
+	cfg moduletools.ClassConfig
+}
+
+func NewClassSettings(cfg moduletools.ClassConfig) *classSettings {
+	return &classSettings{cfg: cfg, BaseClassSettings: *basesettings.NewBaseClassSettings(cfg, false)}
+}
+
+func (cs *classSettings) Model() string {
+	return cs.BaseClassSettings.GetPropertyAsString("model", DefaultWeaviateModel)
+}
+
+func (cs *classSettings) Truncate() string {
+	return cs.BaseClassSettings.GetPropertyAsString("truncate", DefaultTruncate)
+}
+
+func (cs *classSettings) BaseURL() string {
+	return cs.BaseClassSettings.GetPropertyAsString("baseURL", DefaultBaseURL)
+}
+
+func (cs *classSettings) Dimensions() *int64 {
+	defaultValue := PickDefaultDimensions(cs.Model())
+	return cs.BaseClassSettings.GetPropertyAsInt64("dimensions", defaultValue)
+}
+
+func (cs *classSettings) Validate(class *models.Class) error {
+	if err := cs.BaseClassSettings.Validate(class); err != nil {
+		return err
+	}
+
+	model := cs.Model()
+	if !basesettings.ValidateSetting[string](model, availableWeaviateModels) {
+		return errors.Errorf("wrong Weaviate model name, available model names are: %v", availableWeaviateModels)
+	}
+	truncate := cs.Truncate()
+	if !basesettings.ValidateSetting[string](truncate, availableTruncates) {
+		return errors.Errorf("wrong truncate type, available types are: %v", availableTruncates)
+	}
+
+	dimensions := cs.Dimensions()
+	if dimensions != nil {
+		availableDimensions := availableWeaviateModelsDimensions[model]
+		if !basesettings.ValidateSetting[int64](*dimensions, availableDimensions) {
+			return errors.Errorf("wrong dimensions setting for %s model, available dimensions are: %v", model, availableDimensions)
+		}
+	}
```

Comment:
for now we can leave the validation but I'm now leaning more into not having this bc in the future if we will add a new model to the list we would have to add the model to the module, then make a release and then it could be used, this would add another layer of maintainability which I think we could avoid, but if you think it's valuable to have this kind of validation in place, then please leave it